src/odemis/acq/stream/_projection.py

```diff
@@ -218,6 +223,437 @@ def _updateImage(self):
+        # still running this method, the dict might get new entries again, though it should be empty.
+        polarimetry_dict = self._polarimetry_dict
+        polarimetry_dict_raw = self._polarimetry_dict_raw
+        # Note: Method needs about 4sec to display the image for selecting a new ebeam position
+        if pos in polarimetry_dict:
+            stokes_data = polarimetry_dict[pos]
+        else:
+            ebeam_pos = pos[:2]
+            # Compute the polarimetry representation
+            data_raw = self.stream._pos_polarimetry[ebeam_pos]  # get the 6 images for requested ebeam pos
+            try:
+                # TODO get the raw/bg processed data from polar_dict, as now we do bg subtraction twice
+                data_raw_bg_corr = {}
+                # Correct image for background. It must match the polarization (defaulting to MD_POL_NONE).
+                for key in data_raw.keys():
+                    data_raw_bg_corr[key] = self._processBackground(data_raw[key],
+                                                                    data_raw[key].metadata.get(model.MD_POL_MODE,
+                                                                                               model.MD_POL_NONE))
+
+                    if numpy.prod(data_raw_bg_corr[key].shape) > (1280 * 1080):
+                        # TODO not sure this is still true after we changed the triangulation code
+                        # AR conversion fails with very large images due to too much
+                        # memory consumed (> 2Gb). So, rescale + use a "degraded" type that
+                        # uses less memory. As the display size is small (compared
+                        # to the size of the input image, it shouldn't actually
+                        # affect much the output.
+                        logging.info("AR image is very large %s, will convert to "
+                                     "azimuthal projection in reduced precision.",
+                                     data_raw_bg_corr[key].shape)
+                        y, x = data_raw_bg_corr[key].shape
+                        if y > x:
+                            small_shape = 1024, int(round(1024 * x / y))
+                        else:
+                            small_shape = int(round(1024 * y / x)), 1024
+                        # resize
+                        data_raw_bg_corr[key] = img.rescale_hq(data_raw_bg_corr[key], small_shape)
+
+                # convert data into rectangular format (theta-phi-representation)
+                # Check if rectangular converted representation already was calculated for requested ebeam pos.
+                # Note: This calc is very time consuming. Takes about 3.6 sec for one ebeam pos (conversion of 6 images)
+                # and tested on an image of size (256, 1024)
+                try:
+                    polarimetry_dict_raw[ebeam_pos]
```

Comment:
That looks like a complicated way to write `if ebeam_pos not in polarimetry_dict_raw:` ;-)

Reply:
True :D